src/core/mac/mac.cpp

```diff
@@ -1650,6 +1771,12 @@ void Mac::HandleReceivedFrame(RxFrame *aFrame, otError aError)
 
     error = ProcessReceiveSecurity(*aFrame, srcaddr, neighbor);
 
+#if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+    if (aFrame->GetVersion() == Frame::kFcfFrameVersion2015)
+    {
+        ProcessCsl(*aFrame, srcaddr);
+    }
+#endif
```

Comment:
Put this after line 1806? I think for received frame, we should check the MIC and security is passed before the CSL IE is processed.